File: env.ts
import "https://deno.land/x/dotenv/load.ts";
import { cleanEnv, num, str } from "https://deno.land/x/envalid/mod.ts";

export default cleanEnv(Deno.env.toObject(), {
  BOT_TOKEN: str(),
  BOT_USERNAME: str(),
  CACHE_CHAT_ID: num(),
  CONNECTION_URL: str(),
  CHROMIUM_PATH: str(),
  DATABASE_FILE: str({ default: "database.sqlite" }),
  APP_ID: num(),
  APP_SECRET: str(),
  APP_REDIR_URL: str(),
  APP_PERMISSIONS: str(),
});

File: env/env.go
package env

import (
	"os"
	"strconv"

	"github.com/joho/godotenv"
)

var (
	AppID          string
	AppPermissions string
	AppRedirectURL string
	AppSecret      string
	BotToken       string
	BotUsername    string
	CacheChatID    int64
	ConnectionURL  string
	ServerAddress  string
)

func Initialize() error {
	err := godotenv.Load()
	if err != nil {
		return err
	}
	AppID = os.Getenv("APP_ID")
	AppPermissions = os.Getenv("APP_PERMISSIONS")
	AppRedirectURL = os.Getenv("APP_REDIRECT_URL")
	AppSecret = os.Getenv("APP_SECRET")
	BotToken = os.Getenv("BOT_TOKEN")
	BotUsername = os.Getenv("BOT_USERNAME")
	CacheChatID, err = strconv.ParseInt(os.Getenv("CACHE_CHAT_ID"), 10, 64)
	if err != nil {
		return err
	}
	ConnectionURL = os.Getenv("CONNECTION_URL")
	ServerAddress = os.Getenv("SERVER_ADDRESS")
	return nil
}
